pilot/pkg/features/pilot.go

```diff
@@ -646,6 +646,10 @@ var (
 		"If enabled, the cluster secret watcher will watch the namespace of the external cluster instead of config cluster").Get()
 
 	SidecarIgnorePort = env.Register("SIDECAR_IGNORE_PORT_IN_HOST_MATCH", true, "If enabled, port will not be used in vhost domain matches.").Get()
+
+	EnableConfigurationDiscovery = env.Register("ENABLE_DISCOVERY_SELECTOR_FOR_CUSTOM_RESOURCE", false,
```

Comment:
Since this feature flag is going to be used generically for other resource scoping as well as mentioned in the RFC, would it make sense to rename it in a bit more generic way?

Reply:
+1, but we should also make all the changes we are going to make in 1 release - so there is only 1 user facing behavior change.

Reply:
updated